File: typescript/src/services/mfa-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { DisableMfaDto, EnableMfaDto, MfaDto, RegenerateBackupCodesDto } from '../models/index.js';


export class MFAApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async mfaSetup(): Promise<ApiResponse<unknown>> {
    return this.http.post(`mfa/setup`, undefined);
  }

  async mfaEnable(request: EnableMfaDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`mfa/enable`, request);
  }

  async mfaDisable(request: DisableMfaDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`mfa/disable`, request);
  }

  async mfaStatus(): Promise<ApiResponse<unknown>> {
    return this.http.post(`mfa/status`, undefined);
  }

  async mfaRegenerateBackupCodes(request: RegenerateBackupCodesDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`mfa/generate-backup-codes`, request);
  }

  async mfaVerify(request: MfaDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`mfa/verify`, request);
  }

  async mfaValidateAccess(): Promise<ApiResponse<unknown>> {
    return this.http.post(`mfa/validate-access`, undefined);
  }

}

File: c#/src/Docmost.Sdk/Services/MFAApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class MFAApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> MfaSetupAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("mfa/setup", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> MfaEnableAsync(EnableMfaDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("mfa/enable", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> MfaDisableAsync(DisableMfaDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("mfa/disable", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> MfaStatusAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("mfa/status", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> MfaRegenerateBackupCodesAsync(RegenerateBackupCodesDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("mfa/generate-backup-codes", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> MfaVerifyAsync(MfaDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("mfa/verify", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> MfaValidateAccessAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("mfa/validate-access", null, cancellationToken).ConfigureAwait(false);
    }

}
